Signal/src/ViewControllers/ConversationView/Cells/TypingIndicatorCell.swift

```diff
@@ -0,0 +1,154 @@
+                bubbleView.autoPinLeading(toTrailingEdgeOf: avatarView, offset: kAvatarHSpacing),
+                bubbleView.autoAlignAxis(.horizontal, toSameAxisOf: avatarView)
+                ])
+
+        } else {
+            avatarView.removeFromSuperview()
+        }
+    }
+
+    private func configureAvatarView() -> UIView? {
+        guard let viewItem = self.viewItem else {
+            owsFailDebug("Missing viewItem")
+            return nil
+        }
+        guard let typingIndicators = viewItem.interaction as? TypingIndicatorInteraction else {
+            owsFailDebug("Missing typingIndicators")
+            return nil
+        }
+        guard shouldShowAvatar() else {
+            return nil
+        }
+        guard let colorName = viewItem.authorConversationColorName else {
+            owsFailDebug("Missing authorConversationColorName")
+            return nil
+        }
+        guard let authorAvatarImage =
+            OWSContactAvatarBuilder(signalId: typingIndicators.recipientId,
+                                    colorName: ConversationColorNameForString(colorName),
+                                    diameter: UInt(kAvatarSize)).build() else {
+                                        owsFailDebug("Could build avatar image")
+                                        return nil
+        }
+        avatarView.image = authorAvatarImage
+        return avatarView
+    }
+
+    private func shouldShowAvatar() -> Bool {
+        guard let viewItem = self.viewItem else {
+            owsFailDebug("Missing viewItem")
+            return false
+        }
+        return viewItem.isGroupThread
+    }
+
+    @objc
+    public override func cellSize() -> CGSize {
+        guard let conversationStyle = self.conversationStyle else {
+            owsFailDebug("Missing conversationStyle")
+            return .zero
+        }
+
+        let insetsSize = CGSize(width: conversationStyle.textInsetHorizontal * 2,
+                                height: conversationStyle.textInsetTop + conversationStyle.textInsetBottom)
+        let typingIndicatorSize = typingIndicatorView.sizeThatFits(.zero)
+        let bubbleSize = CGSizeAdd(insetsSize, typingIndicatorSize)
+
+        if shouldShowAvatar() {
+            let avatarSize = CGSize(width: kAvatarSize, height: kAvatarSize)
+            return CGSizeCeil(CGSize(width: avatarSize.width + kAvatarHSpacing + bubbleSize.width,
+                                     height: max(avatarSize.height, bubbleSize.height)))
+        } else {
+            return bubbleSize
+        }
+    }
+
+    @objc
+    public override func prepareForReuse() {
+        super.prepareForReuse()
+
+        NSLayoutConstraint.deactivate(viewConstraints)
+        viewConstraints = [NSLayoutConstraint]()
+
+        avatarView.image = nil
+        avatarView.removeFromSuperview()
+
+        typingIndicatorView.stopAnimation()
+
+        NotificationCenter.default.removeObserver(self)
```

Comment:
What notifications are we no longer observing here?

Reply:
This is obsolete, removed.